src/App.js

```diff
@@ -0,0 +1,106 @@
+import React, {useState} from 'react';
+import Screen from './components/Screen/Screen'
+import Button from './components/Button/Button'
+import './App.css';
+
+function App() {
+  const [screenText, setScreen] = useState('0')
+  function handleClick(text) {
+    if (text === '=') {
+      const screenSplit = screenText.split(' ')
+      if (screenSplit.length <= 2) {
+        setScreen('error')
+      } else if (screenSplit.length > 3) {
+        setScreen('error')
+      } else if (screenSplit[1] === '÷') {
+        setScreen((screenSplit[0] / screenSplit[2]).toString())
+      } else if (screenSplit[1] === '×') {
+        setScreen((screenSplit[0] * screenSplit[2]).toString())
+      } else if (screenSplit[1] === '-') {
+        setScreen((parseFloat(screenSplit[0]) - parseFloat(screenSplit[2])).toString())
+      } else if (screenSplit[1] === '+') {
+        setScreen((parseFloat(screenSplit[0]) + parseFloat(screenSplit[2])).toString())
+      } else {
+        setScreen('error')
+      }
+    } else if (screenText === '0' && text !== '⇐') {
+      setScreen(text)
+    } else if (text === '⇐' || text === 'clr') {
+      if (screenText.length === 1 || screenText === 'error' || text === 'clr') {
+        setScreen('0')
+      } else {
+        setScreen(screenText.substring(0, screenText.length - 1))
+      }
```

Comment:
- Consider using a `switch` statement instead of all the `esle if`s, also you could make this into a helper function and just call it with arguments it would make the code readable.

Reply:
Agree with @Redvanisation  above. I would even declare rules separately in order to make this work.

```
// example only
const RULES = [
  {
     matcher: (text) => text === "+", 
     callback: (a,b) => a + b,
  },
  {
     matcher: (text) => text === "-", 
     callback: (a,b) => a - b,
  },
];

const text = "+", 
  a = 1,
  b = 2;

// find the first rule that matches and perform the calculation
const result = RULES.first(({ matcher }) => matcher(text)).callback(a, b);
```

Reply:
The above can be extended with nested matches if needs be in the future.